extensions/wikia/ARecoveryEngine/js/helper.js

```diff
@@ -81,12 +81,31 @@ define('ext.wikia.aRecoveryEngine.recovery.helper', [
 		}
 	}
 
+	function getSafeUri(url) {
+		return win._sp_.getSafeUri(url);
+	}
+
+	function isSourcePointResultDefined() {
+		return win.ads && win.ads.runtime.sp && win.ads.runtime.sp.blocking !== undefined;
+	}
+
+	function runAfterDetection(callback) {
+		if (isSourcePointResultDefined()) {
```

Comment:
You can achieve the same thing without checking current status of detection. `addOnBlockingCallback` is using lazy queue and for sure it's fired once. It would be beneficial to add second queue for `sp.not_blocking` and reuse it in the same way